File: backend/app/schemas/poverty.py
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel, Field


class PovertyCountyOut(BaseModel):
    province: str = Field(..., description="省份")
    city: Optional[str] = Field(default=None, description="市/州")
    county: str = Field(..., description="县/区")

    population: Optional[int] = Field(default=None, description="人口")
    income_per_capita_yuan: Optional[float] = Field(default=None, description="人均收入(元)")
    energy_condition: Optional[str] = Field(default=None, description="能源条件/电网情况")
    tags: Optional[str] = Field(default=None, description="标签")
    adcode: Optional[str] = Field(default=None, description="行政区划代码")

    source: str = Field(..., description="数据源")
    source_url: Optional[str] = Field(default=None, description="源链接")

File: frontend/src/api/poverty.ts
import request from '@/api/request';

export interface PovertyCountyItem {
  province: string;
  city?: string | null;
  county: string;
  population?: number | null;
  income_per_capita_yuan?: number | null;
  energy_condition?: string | null;
  tags?: string | null;
  adcode?: string | null;
  source: string;
  source_url?: string | null;
}

export function getPovertyCountiesApi(params?: {
  province?: string;
  city?: string;
  skip?: number;
  limit?: number;
}): Promise<PovertyCountyItem[]> {
  return request({
    url: '/poverty/counties',
    method: 'get',
    params
  }) as Promise<PovertyCountyItem[]>;
}
